### PR TITLE
kernel: change argument validation for OBJ_{SET,MAP} to use RequireArgumentCondition

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: src/objset.c [hunk 1]
```diff
@@ -844,9 +835,13 @@ static Obj FuncCLEAR_OBJ_SET(Obj self, Obj set) {
...
  */
 
-static Obj FuncOBJ_SET_VALUES(Obj self, Obj set) {
-  CheckArgument("OBJ_SET_VALUES", set, T_OBJSET, -1);
-  return ObjSetValues(set);
+static Obj FuncOBJ_SET_VALUES(Obj self, Obj set)
+{
+    RequireArgumentCondition("OBJ_SET_VALUES", set, "objset",
+                             TNUM_OBJ(set) == T_OBJSET,
+                             "must be a mutable object set");
```
Comment (fingolfin) [inline src/objset.c]: @rbehrends is there any reason this function requires mutable objset?

File: src/objset.c [hunk 2]
```diff
@@ -967,9 +982,13 @@ static Obj FuncCLEAR_OBJ_MAP(Obj self, Obj map) {
...
  */
 
-static Obj FuncOBJ_MAP_VALUES(Obj self, Obj map) {
-  CheckArgument("OBJ_MAP_VALUES", map, T_OBJMAP, -1);
-  return ObjMapValues(map);
+static Obj FuncOBJ_MAP_VALUES(Obj self, Obj map)
+{
+    RequireArgumentCondition("OBJ_MAP_VALUES", map, "objmap",
+                             TNUM_OBJ(map) == T_OBJMAP,
+                             "must be a mutable object map");
```
Comment (fingolfin) [inline src/objset.c]: @rbehrends is there any reason this function requires mutable objmap?

File: src/objset.c [hunk 3]
```diff
@@ -980,9 +999,13 @@ static Obj FuncOBJ_MAP_VALUES(Obj self, Obj map) {
...
  */
 
-static Obj FuncOBJ_MAP_KEYS(Obj self, Obj map) {
-  CheckArgument("OBJ_MAP_KEYS", map, T_OBJMAP, -1);
-  return ObjMapKeys(map);
+static Obj FuncOBJ_MAP_KEYS(Obj self, Obj map)
+{
+    RequireArgumentCondition("OBJ_MAP_KEYS", map, "objmap",
+                             TNUM_OBJ(map) == T_OBJMAP,
+                             "must be a mutable object map");
```
Comment (fingolfin) [inline src/objset.c]: @rbehrends is there any reason this function requires mutable objmap?

**Comment by ChrisJefferson** [comment]
I'm happy with this, we can always change the requirements on mutability later, if/when we find out more.

File: src/objset.c (hunk 1 above)
Comment (rbehrends) [reply src/objset.c]: No, this and the other checks are probably the result of copy-and-paste errors.

**Comment by fingolfin** [comment]
Since @rbehrends clarified now, I'll update this PR accordingly.

**Pull request merged**